.github/workflows/prereq.yml

```diff
@@ -18,13 +20,24 @@ jobs:
           fetch-depth: 500
       - name: Environment
         run: |
-          echo $GITHUB_REF_NAME
-          git rev-list -1 HEAD
-          git rev-list -1 HEAD^1
-          git rev-list -1 HEAD^2
+          echo "GITHUB_REF_NAME: $GITHUB_REF_NAME"
+          echo "GITHUB_BASE_REF: $GITHUB_BASE_REF"
+          echo "HEAD: $(git rev-list -1 HEAD)"
+          if test -n "$GITHUB_BASE_REF"; then
+            BASEB=$(git rev-list -1 HEAD^1)
+            HEADB=$(git rev-list -1 HEAD^2)
```

Comment:
If I got it right then I will try to guess that, probably, it was read as `HEAD~1` and `HEAD~2`.

Upon pull request GitHub runs the job against a merge commit, where the first parent `HEAD^1` is `$GITHUB_BASE_REF` (e.g. freebsd/freebsd-src/main branch) and the second parent `HEAD^2` is `$GITHUB_HEAD_REF` (e.g. contributor/freebsd-src/some-fix branch). The reasoning details can be found here: https://reviews.freebsd.org/D40261#916852.
With fetch-depth=500 we get both parents 500 commits deep, i.e. both base and head branches are already fetched by actions/checkout. The only nuance is that `$GITHUB_BASE_REF` and `$GITHUB_HEAD_REF` references are not present within a runner's git repo. The only reference we have is named something like `refs/remotes/pull/<PRID>/merge` and points to this merge commit. We could invoke additional git fetch, but it won’t bring anything new, both branches' heads are already fetched, they are `HEAD^1` and `HEAD^2`.

This PR #751 is the test itself to show how it works: https://github.com/freebsd/freebsd-src/actions/runs/5082844700/jobs/9133072060.

Yes, it may look tricky from `prereq.yml` reader perspective. Hence, it looks like we have at least two options: leave it shorter without additional git fetch commands (maybe with some comment added), or additionally invoke git fetch `$GITHUB_BASE_REF` to make it more obvious. I think your experience has better vision which way fits.